airflow/models/dag.py

```diff
@@ -403,11 +403,12 @@ def __init__(
         # set timezone from start_date
         tz = None
         if start_date and start_date.tzinfo:
-            tz = pendulum.instance(start_date).timezone
+            tz = pendulum.instance(start_date, tz=start_date.tzinfo or settings.TIMEZONE).timezone
```

Comment:
Nice. Sounds like an actual improvement of behaviour (bug fix)? Should we change the title of the PR/commit to make it clearer in the changelog (or maybe separate those out)?

Reply:
I don't _think_ this changes the behaviour, not for anything but obscure edge cases anyway.

Reply:
Fine for me. It's just an annoying feeling that some of those edge cases might fix some "real errors" :) and just indication in the commit message that it could improve things might be good for commit message in this case:

"Improve handling edge-cases in airlfow models by applying mypy suggestions", maybe :)